src/main.rs

```diff
@@ -178,16 +180,45 @@ fn start_zenith(
         };
 
         debug!("Create Renderer");
-        let mut r = TerminalRenderer::new(
-            rate,
-            cpu_height as i16,
-            net_height as i16,
-            disk_height as i16,
-            process_height as i16,
-            sensor_height as i16,
-            graphics_height as i16,
-            db,
-        );
+
+        let mut geometry: Vec<(Section, f64)> = Vec::new();
+        if cpu_height > 0 {
+            geometry.push((Section::CPU, cpu_height as f64));
+        }
+        if net_height > 0 {
+            geometry.push((Section::Network, net_height as f64));
+        }
+        if disk_height > 0 {
+            geometry.push((Section::Disk, disk_height as f64));
+        }
+        if graphics_height > 0 {
+            geometry.push((Section::Graphics, graphics_height as f64));
+        }
+        if process_height > 0 {
+            geometry.push((Section::Process, process_height as f64));
+        }
+        assert_eq!(sensor_height, 0); // not implemented
+
+        let num_sections = geometry.len();
+        if num_sections == 0 {
+            panic!("All sections have size specified as zero!");
+        }
+        // sum of minimum percentages should not exceed 100%
+        let sum_heights = sum_section_heights(&geometry);
+        if sum_heights > 100.1 {
+            panic!(
```

Comment:
Can you change the `panic!` for incorrect user specified parameters to use `println` and `exit`

Reply:
Shouldn't it be fine here since panic_hook has already been setup. Or you want it to be so after refactor so that it can be invoked from main?

Reply:
I'm not sure it's relevant anymore but I just thought it was bad form to present a panic to the user when what we really want to do is let them know the parameters they entered are in conflict.